lib/src/components/NxCombobox/__tests__/NxCombobox.test.tsx

```diff
@@ -0,0 +1,522 @@
+              inputElement = getByRole('combobox'),
+              optionBtn = getByRole('option', { hidden: true });
+
+          inputElement.focus();
+          await userEvent.keyboard('[ArrowDown]');
+          expect(optionBtn).toHaveAttribute('aria-selected', 'true');
+          await userEvent.keyboard('[Home]o');
+          expect(onChange).toBeCalledWith('of');
+          expect(optionBtn).toHaveAttribute('aria-selected', 'false');
+        });
+
+    it('places the editing cursor at the end of the input field when End key is pressed with focus in dropdown',
+        async function() {
+          const onChange = jest.fn(),
+              { getByRole } = quickRender({ matches: [{ id: '1', displayName: 'Foo' }], value: 'f', onChange }),
+              inputElement = getByRole('combobox'),
+              optionBtn = getByRole('option', { hidden: true });
+
+          inputElement.focus();
+          await userEvent.keyboard('[ArrowUp]');
+          expect(optionBtn).toHaveAttribute('aria-selected', 'true');
+          await userEvent.keyboard('[Home][End]o');
+          expect(onChange).toBeCalledWith('fo');
+          expect(optionBtn).toHaveAttribute('aria-selected', 'false');
+        });
+
+    it('should open dropdown and move visual focus to the first option with aria-selected set to true' +
+      'if dropdown is closed and there are matches when down arrow key is pressed', async function() {
+      const { getByRole, getAllByRole } = quickRender({
+            matches: [{ id: '1', displayName: 'Foo' }, { id: '2', displayName: 'Boo' }] }),
+          inputElement = getByRole('combobox'),
+          dropdownElement = getByRole('listbox', { hidden: true }),
+          firstOptBtn = getAllByRole('option', { hidden: true })[0];
+
+      expect(dropdownElement).toHaveAttribute('aria-hidden', 'true');
+      inputElement.focus();
+      await userEvent.keyboard('[ArrowDown]');
+      expect(dropdownElement).toHaveAttribute('aria-hidden', 'false');
+      expect(firstOptBtn).toHaveAttribute('aria-selected', 'true');
+    });
+
+    it('should move visual focus to the next option if dropdown is open' +
+      'and the down arrow key is pressed', async function() {
+      const { getByRole, getAllByRole } = quickRender({
+            matches: [{ id: '1', displayName: 'Foo' }, { id: '2', displayName: 'Boo' }] }),
+          inputElement = getByRole('combobox'),
+          firstOptBtn = getAllByRole('option', { hidden: true })[0],
+          secondOptBtn = getAllByRole('option', { hidden: true })[1];
+
+      inputElement.focus();
+      await userEvent.keyboard('[ArrowDown]');
+      expect(firstOptBtn).toHaveAttribute('aria-selected', 'true');
+      await userEvent.keyboard('[ArrowDown]');
+      expect(firstOptBtn).toHaveAttribute('aria-selected', 'false');
+      expect(secondOptBtn).toHaveAttribute('aria-selected', 'true');
+    });
+
+    it('should move visual focus to the first option if current focus is on the last option' +
```

Comment:
All of these tests about navigating the dropdown with the keyboard should probably check `aria-activedescendant` in addition to `aria-selected`.

Reply:
Updated here 6dbc601488edec3acdf06ca1c289688de300a1a8.